cdap-event-reader-spi/src/main/java/io/cdap/cdap/spi/events/ReceivedEvent.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.spi.events;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ *  Incoming PubSub Message that contains an event.
+ */
+public class ReceivedEvent {
```

Comment:
Having both StartProgramEvent and ReceivedEvent classes are very confusing. Specially, since the two do not extend each other.